spec/unit/provider/mount/linux_spec.rb

```diff
@@ -0,0 +1,91 @@
+require "spec_helper"
+require "ostruct"
+
+describe Chef::Provider::Mount::Linux do
+  before(:each) do
+    @node = Chef::Node.new
+    @events = Chef::EventDispatch::Dispatcher.new
+    @run_context = Chef::RunContext.new(@node, {}, @events)
+
+    @new_resource = Chef::Resource::Mount.new("/tmp/foo")
+    @new_resource.device      "/dev/sdz1"
+    @new_resource.device_type :device
+    @new_resource.fstype      "ext3"
+
+    @new_resource.supports remount: false
+
+    @provider = Chef::Provider::Mount::Linux.new(@new_resource, @run_context)
+
+    allow(::File).to receive(:exists?).with("/dev/sdz1").and_return true
+    allow(::File).to receive(:exists?).with("/tmp/foo").and_return true
+    allow(::File).to receive(:realpath).with("/dev/sdz1").and_return "/dev/sdz1"
+    allow(::File).to receive(:realpath).with("/tmp/foo").and_return "/tmp/foo"
+  end
```

Comment:
```suggestion

  let(:run_context) do
    node = Chef::Node.new
    events = Chef::EventDispatch::Dispatcher.new
    run_context = Chef::RunContext.new(node, {}, events)
  end

  let(:new_resource) do
    new_resource = Chef::Resource::Mount.new("/tmp/foo")
    new_resource.device      "/dev/sdz1"
    new_resource.device_type :device
    new_resource.fstype      "ext3"
    new_resource.supports remount: false
    new_resource
  end

  let(:provider) do
    Chef::Provider::Mount::Linux.new(@new_resource, @run_context)
  end

  before(:each) do
    allow(::File).to receive(:exists?).with("/dev/sdz1").and_return true
    allow(::File).to receive(:exists?).with("/tmp/foo").and_return true
    allow(::File).to receive(:realpath).with("/dev/sdz1").and_return "/dev/sdz1"
    allow(::File).to receive(:realpath).with("/tmp/foo").and_return "/tmp/foo"
  end
```

Using instance variables in tests is not a good pattern, we should use rspec stuff for this.

Reply:
Agreed, I have updated the changes.